### PR TITLE
Make HTTPFileLoader more intelligent (Fixes #128)

### LINKED ISSUE #128: Improve remote file loading to use the expected file hint
When invoking the `NewFileLoader` function we have to pass the root of the repository, the path or URI of the remote file and a hint about what kind of file we are looking for. We shouldn't need to pass the root of the repository code since the caller should know if it's a path to the local filesystem or a URI. The `httpFileLoader` should also check if the URI is a pointing to the hint file and if not it should be appended. This behaviour resembles the way `kustomize` resolves directories.

### PR DESCRIPTION
**Pull request by ptzianos**
Splits away the Loader and its implementations into the pkg/files module and adds more logic to the HTTPFileLoader so that it can append the target file to the URL if it's missing. Tests are also added to make sure that the HTTPFileLoader is 100% covered now.
A new interface and a mock implementation has been added for HTTP requests.

### CODE REVIEW

**Review by northdpole — APPROVED** [review]

File: pkg/manifests/tekton.go [hunk 1]
```diff
@@ -4,29 +4,31 @@ import (
 	"context"
 	"fmt"
 
-	tektonV1Beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
 )
 
-func LoadTektonV1Beta1Task(ctx context.Context, root, pathOrURI string) (*tektonV1Beta1.Task, error) {
-	obj, gKV, err := LoadK8sManifest(ctx, root, pathOrURI, "task.yaml")
+// LoadTektonV1Beta1Task loads a tektonV1Beta1 Task object from a path
```
[the remark this answers is not in the document]
Comment (ptzianos) [reply pkg/manifests/tekton.go:10]: This is an erroneous finding

File: pkg/manifests/tekton.go [hunk 2]
```diff
@@ -4,29 +4,31 @@
...
```
[the remark this answers is not in the document]
Comment (ptzianos) [reply pkg/manifests/tekton.go:10]: This is an erroneous finding

File: pkg/manifests/decoder.go [hunk 3]
```diff
@@ -4,13 +4,15 @@
...
 	"fmt"
 
-	tektonV1Beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
 	batchv1 "k8s.io/api/batch/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/runtime/serializer"
-	jsonSerializer "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	jsonserializer "k8s.io/apimachinery/pkg/runtime/serializer/json"
```
[the remark this answers is not in the document]
Comment (ptzianos) [reply pkg/manifests/decoder.go:12]: This is an erroneous finding

File: pkg/manifests/decoder.go [hunk 4]
```diff
@@ -4,13 +4,15 @@ import (
 	"context"
 	"fmt"
 
-	tektonV1Beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
```
[the remark this answers is not in the document]
Comment (ptzianos) [reply pkg/manifests/decoder.go:7]: This is an erroneous finding

File: pkg/manifests/tekton.go [hunk 5]
```diff
@@ -4,29 +4,31 @@ import (
 	"context"
 	"fmt"
 
-	tektonV1Beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
```
[the remark this answers is not in the document]
Comment (ptzianos) [reply pkg/manifests/tekton.go:7]: This is an erroneous finding

File: pkg/pipelines/tektonV1Beta1.go [hunk 6]
```diff
@@ -118,8 +118,8 @@ func addAnchorParameter(task *tektonV1Beta1API.Task) {
 	})
 }
 
-// ResolveKustomizationResourceBases checks the resources section to find the base pipeline and
-// task and fetches them wherever they are located.
+// ResolveKustomizationResources checks the resources section to find the base pipeline and
```
[the remark this answers is not in the document]
Comment (ptzianos) [reply pkg/pipelines/tektonV1Beta1.go:121]: This is an erroneous finding

File: cmd/draconctl/pipelines/build.go [hunk 7]
```diff
@@ -7,8 +7,9 @@ import (
 	"strings"
 
 	"github.com/spf13/cobra"
-	kustomizeType "sigs.k8s.io/kustomize/api/types"
+	kustomizetypes "sigs.k8s.io/kustomize/api/types"
```
[the remark this answers is not in the document]
Comment (ptzianos) [reply cmd/draconctl/pipelines/build.go:10]: This is an erroneous finding

**Pull request merged**